agents/World.java

```diff
@@ -41,6 +56,18 @@ private static void iterateCSV(Path path, Consumer<CSVRecord> action) throws IOE
         return continents.stream().flatMap(Continent::getCountries).flatMap(Country::getCities);
     }
 
+    public List<String> getFemaleForenames() {
+        return Collections.unmodifiableList(femaleForenames.stream().map(World.FemaleForename::getValue).collect(toList()));
```

Comment:
We want to avoid the stream and collect! This duplicates the list every iteration potentially for every city (and for every agent that uses it). Instead of using a domain object for the forename, I'd suggest:

```
private final List<String> femaleForenames;
...
public World(...) {
    ...
    List<String> femaleForenames = new ArrayList<>();
    iterateCSV(femaleForenamesPath, r -> femaleForenames.add(r.get(0)));
    this.femaleForenames = Collections.unmodifiableList(femaleForenames); // Saving an unmodifiable collection to a final field is the best guarantee of immutability we can achieve.
    ...
}
...
public List<String> getFemaleForenames() {
    return femaleForenames;
}
```

But if we want to continue using the domain object, `getFemaleForenames()` should return `List<FemaleForename>` and the agent should perform `femaleForenames.get(...).getValue();`

Reply:
Aha! Very good point, the changes I made before didn't actually remedy this did they...

I like your suggestion, implementing now.